### DISCUSSION TITLE
Nangate45 netlist does not work in simulation, while other PDKs work without a problem

### DISCUSSION THREAD

**Discussion by Pedniel** [The-OpenROAD-Project/OpenROAD-flow-scripts#2621] [Q&A]
I am writing my bachelor thesis with a focus on PDK differences and i'm currently trying to simulate the example aes design provided by the OpenROAD-flow-scripts repo.

I already synthesized the aes with Skywater130, Asap7 and ihp-sg13g2 and managed to get all of them simulated with a back annotated sdf file, using xcelium. I simulate the aes netlist using a customized version of the testbench of the same aes found on Opencores.
However, I cant get the nangate45 netlist to give me a proper simulation result, even without back annotated delays.

This is the result of a working simulation using asap7:

DATA: 6bc1bee22e409f96e93d7e117393172a
KEY: 2b7e151628aed2a6abf7158809cf4f3c

Result : 3ad77bb40d7a3660a89ecaf32466ef97

This is the same simulation result with nangate45:

DATA: 6bc1bee22e409f96e93d7e117393172a
KEY: 2b7e151628aed2a6abf7158809cf4f3c

Result : xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

I also ran a simple power analysis on all of these designs and while all other design give expected results, the nangate45 netlist does not.
All the other designs have the expected 10 large toggle spikes for the 10 aes encryption rounds of a 128 bit aes but the nangate45 netlist only has two clock periods with about 2500 toggles and then barely any toggles at all, which is weird. 
Sadly I cannot provide images of the SPA, since I do not own the used SPA tool.

Could anyone kindly assist me with this problem? Did someone maybe manage to get the example aes simulated with nangate45? I'm not sure if my simulation flow is at fault here, since it works with the other PDKs, but i'm open to suggestions. 

EDIT: GF180 also works without a problem, its just nangate45 thats not working.

**Comment by rovinski** [comment]
This is more of a logic simulation question, which is not particularly well-suited here. I would suggest asking on the open-source silicon slack or other communities which are focused on logic simulation.

As for debugging, maybe start by backtracing the Xs and figure out where they start propagating from. It could be that there's a difference between the way the netlist is generated for Nangate45 (which is a synthetic PDK and is missing several cells that the other PDKs mentioned have) and the way it was generated for the other PDKs.
